src/gasPriceOracle/adapters/ethereum.ts

```diff
@@ -3,63 +3,87 @@ import { providers } from "ethers";
 import { BigNumber, bnZero, getNetworkName } from "../../utils";
 import { GasPriceEstimate } from "../types";
 import { gasPriceError } from "../util";
+import { GasPriceEstimateOptions } from "../oracle";
 
 /**
+ * @dev If GAS_PRICE_EIP1559_RAW_${chainId}=true, then constructs total fee by adding
+ * eth_getBlock("pending").baseFee to eth_maxPriorityFeePerGas, otherwise calls the ethers provider's
+ * getFeeData() method which adds eth_getBlock("latest").baseFee to a hardcoded priority fee of 1.5 gwei.
  * @param provider ethers RPC provider instance.
- * @param chainId Chain ID of provider instance.
  * @returns Promise of gas price estimate object.
  */
-export function eip1559(provider: providers.Provider, chainId: number): Promise<GasPriceEstimate> {
-  const useRaw = process.env[`GAS_PRICE_EIP1559_RAW_${chainId}`] === "true";
-  return useRaw ? eip1559Raw(provider, chainId) : eip1559Bad(provider, chainId);
+export function eip1559(provider: providers.Provider, opts: GasPriceEstimateOptions): Promise<GasPriceEstimate> {
+  const useRaw = process.env[`GAS_PRICE_EIP1559_RAW_${opts.chainId}`] === "true";
```

Comment:
Should we not make `useRaw` the default for all chains...?

Reply:
Talked with Paul offline about this and the thinking is, we want to make this "raw" `eip1559()` function the default and only option eventually, but we also want to slowly roll it out and test it in production. So gate-keeping it with an environment variable is the easiest way to slowly roll it out. 35a0420bd4c5f1b1782d6cdbf29fa12bcf5820d0